pkgs/top-level/all-packages.nix

```diff
@@ -1049,6 +1049,9 @@ with pkgs;
   kitty = callPackage ../applications/terminal-emulators/kitty {
     harfbuzz = harfbuzz.override { withCoreText = stdenv.isDarwin; };
     inherit (darwin.apple_sdk.frameworks) Cocoa CoreGraphics Foundation IOKit Kernel OpenGL;
+    UserNotifications = if (builtins.hasAttr "UserNotifications" darwin.apple_sdk.frameworks)
+      then darwin.apple_sdk.frameworks.UserNotifications
+      else null;
```

Comment:
There needs to be another way than this hackery. Could we gate the usage of the framework behind a lib.optional?

Reply:
In fact, I tried this first:

```
} // lib.optionalAttrs (builtins.hasAttr "UserNotifications" darwin.apple_sdk.frameworks) {
    UserNotifications = darwin.apple_sdk.frameworks.UserNotifications;
};
```

But it does not work, see https://github.com/NixOS/nixpkgs/pull/137512#issuecomment-961899737 for log.


`pinentry` in pkgs/top-level/all-packages.nix has similar logic:
```
pinentry = libsForQt5.callPackage ../tools/security/pinentry {
  libcap = if stdenv.isDarwin then null else libcap;
};
```

Reply:
> pinentry in pkgs/top-level/all-packages.nix has similar logic:

I am going to change that right now because that is a really error prone coding style.